packages/core/contracts/financial-templates/common/WETH9.sol

```diff
@@ -80,7 +80,7 @@ contract WETH9 {
     ) public returns (bool) {
         require(balanceOf[src] >= wad);
 
-        if (src != msg.sender && allowance[src][msg.sender] != uint256(-1)) {
+        if (src != msg.sender && allowance[src][msg.sender] != type(uint256).max) {
```

Comment:
i just looked up some breaking changes with solidity .8, so this is the recommended way to get `uint256(-1)`, but i have some unrelated questions. 
- if no allowance is set, is `type(uint256).max` what the value defaults to?
- if the max allowance is set, does it differ from `type(uint256).max`

Reply:
this is in the default WETH9 contract so it's not part of our implementation.